.github/workflows/ci.yml

```diff
@@ -136,10 +136,17 @@ jobs:
           python -m pip install dist/*whl
 
       - name: Install Python dependencies
+        if: "(github.event_name == 'push' && !contains(github.ref, 'release')) || (github.event_name == 'pull_request' && !contains(github.base_ref, 'release-6.1.0'))"
         run: |
           python -m pip install -r build_tools/requirements.txt
           python -m pip install build hatchling hatch-build-scripts hatch-requirements-txt hatch-vcs hatch-sphinx
 
+      ### For pull_requests with the release branch as the base -or- branches with the name *release*
+      - name: Install Python dependencies for release branches
+        if: "(github.event_name == 'push' && contains(github.ref, 'release')) || (github.event_name == 'pull_request' && contains(github.base_ref, 'release-6.1.0'))"
+        run: |
+          python -m pip install --no-deps -r build_tools/requirements-release-${{ matrix.os }}.txt
+
```

Comment:
Is there merit in putting these into an environment variable (say, `USE_PINNED_VERSIONS`) so that the calculation only needs to be done in one place and the negation is perhaps easier too? Using that for `SIGN_INSTALLER` felt like it made life easier for those steps.

Reply:
There is, but for this PR, I'm going to skip it.